subgraph/src/mappings/vault.ts

```diff
@@ -1,16 +1,88 @@
-import { BigInt, log } from "@graphprotocol/graph-ts";
-import { DepositBurned, DepositMinted } from "../types/templates/Vault/IVault";
+import { BigDecimal, BigInt, ByteArray, log } from "@graphprotocol/graph-ts";
+import {
+  DepositBurned,
+  DepositMinted,
+  YieldClaimed
+} from "../types/templates/Vault/IVault";
 import {
   Sponsored,
   Unsponsored
 } from "../types/templates/Vault/IVaultSponsoring";
-import { Sponsor, Claimer, Deposit, Foundation, Vault } from "../types/schema";
+import {
+  Sponsor,
+  Claimer,
+  Deposit,
+  Foundation,
+  Vault,
+  Donation
+} from "../types/schema";
+
+export function handleYieldClaimed(event: YieldClaimed): void {
+  const claimerId = event.params.claimerId.toString();
+  const claimer = Claimer.load(claimerId)!;
+  const vault = Vault.load(claimer.vault)!;
+
+  claimer.claimed = claimer.claimed.plus(event.params.amount);
+  claimer.shares = claimer.shares.minus(event.params.burnedShares);
+  vault.totalShares = vault.totalShares.minus(event.params.burnedShares);
+  log.debug("claim, sub shares {}", [event.params.burnedShares.toString()]);
+
+  let totalClaimedShares = new BigInt(0);
+
+  for (let i = 0; i < claimer.depositsIds.length; i++) {
+    const deposit = Deposit.load(claimer.depositsIds[i]);
+
+    if (!deposit) continue;
+
+    const claimedShares = deposit.shares
+      .times(event.params.amount)
+      .div(event.params.burnedShares)
+      .minus(deposit.amount)
+      .times(event.params.burnedShares)
+      .div(event.params.amount);
+
+    totalClaimedShares = totalClaimedShares.plus(claimedShares);
+    deposit.shares = deposit.shares.minus(claimedShares);
+    deposit.save();
+
+    if (
+      event.params.to.equals(
+        ByteArray.fromHexString("0x4940c6e628da11ac0bdcf7f82be8579b4696fa33")
+      ) &&
+      claimedShares.gt(BigInt.fromI32(0))
+    ) {
+      const id =
+        event.transaction.hash.toHex() +
+        "-" +
+        event.logIndex.toString() +
+        "-" +
+        i.toString();
+
+      const donation = new Donation(id);
+      donation.txHash = event.transaction.hash;
+      donation.amount = claimedShares
+        .times(event.params.amount)
+        .div(event.params.burnedShares);
+      donation.owner = deposit.depositor;
+      donation.destination = deposit.data;
+
+      donation.save();
+    }
+  }
+
+  if (!event.params.burnedShares.equals(totalClaimedShares)) {
+    throw "The math for the claimed shares doesn't add up to the total burned shares";
+  }
+
+  claimer.save();
+  vault.save();
+}
 
 export function handleDepositMinted(event: DepositMinted): void {
-  const foundationId = event.params.groupId.toHexString();
-  const depositId = event.params.id.toHexString();
-  const claimerId = event.params.claimerId.toHexString();
-  const vaultId = event.address.toHexString();
+  const foundationId = event.params.groupId.toString();
```

Comment:
@naps62 I just realized that this won't work with multiple vaults. The groupId is an incremented number that will exist in all vaults eventually. I need to use the vault id as well.